skyportal/models.py

```diff
@@ -196,6 +211,33 @@ def desi_dr8_url(self):
         return (f"http://legacysurvey.org/viewer/jpeg-cutout?ra={self.ra}"
                 f"&dec={self.dec}&size=200&layer=dr8&pixscale=0.262&bands=grz")
 
+    def airmass(self, telescope, time):
+        """Return the airmass of the Obj at time `time` from Telescope
+        `telescope`.
+
+        Parameters
+        ----------
+
+        telescope: skyportal.models.Telescope
+            The telescope to use for the airmass calculation
+
+        time: astropy.time.Time
+            The time or times at which to calculate the airmass
+
+        Returns
+        -------
+
+        secz: astropy.units.Quantity
+           The airmass of the Obj at the requested times
+        """
+
+        target = astroplan.FixedTarget(name=self.id, coord=self.skycoord)
+        observer = astroplan.Observer(latitude=telescope.lat * u.deg,
+                                      longitude=telescope.lon * u.deg,
+                                      elevation=telescope.elevation * u.m)
+
+        return observer.altaz(time, target).secz
```

Comment:
Airmass near the horizon isn't well represented by secz (ie. we dont actually live in a plane-parallel atmosphere). Perhaps use Pickering's interpolation? (https://en.wikipedia.org/wiki/Air_mass_(astronomy)#Interpolative_formulas)

Reply:
done